### PR TITLE
Add migration guide for deprecations removed after v2.10

### PR DESCRIPTION
**Pull request by Piinks**
This is an overview of all deprecated API that has been removed since the 2.10 release.

Fixes https://github.com/flutter/flutter/issues/98537

Announcement: https://groups.google.com/g/flutter-announce/c/gq1dAIBXW9A
Quick reference: [flutter.dev/go/deprecations-removed-after-2-10](flutter.dev/go/deprecations-removed-after-2-10)

Related PRs:
- https://github.com/flutter/flutter/pull/98539
- https://github.com/flutter/flutter/pull/98541
- https://github.com/flutter/flutter/pull/98543
- https://github.com/flutter/flutter/pull/98545
- https://github.com/flutter/flutter/pull/98546
- https://github.com/flutter/flutter/pull/98547
- https://github.com/flutter/flutter/pull/98549
- https://github.com/flutter/flutter/pull/98613
- https://github.com/flutter/flutter/pull/98578
- https://github.com/flutter/flutter/pull/98582
- https://github.com/flutter/flutter/pull/98583
- https://github.com/flutter/flutter/pull/98615
- https://github.com/flutter/flutter/pull/98616

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/2-10-deprecations.md [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+deprecated in v1.20, and then extended in v1.26.
+
+They are replaced by new buttons, `TextButton`, `ElevatedButton`, and
+`OutlinedButton`. These new widgets also use new associated themes, rather than
+the generic `ButtonTheme`.
+
+<div class="table-wrapper" markdown="1">
+
+| Old Widget   | Old Theme     | New Widget      | New Theme |
+| -------------|---------------|-----------------|----------|
```
Comment (domesticmouse) [inline src/release/breaking-changes/2-10-deprecations.md:190]: nit: line up |'s for the OCD amongst us plz =)

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
PTAL @sfshaza2 =)

File: src/release/breaking-changes/2-10-deprecations.md (hunk 1 above)
Comment (Piinks) [reply src/release/breaking-changes/2-10-deprecations.md:190]: Happily!

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**